### PR TITLE
Use denser literals for bitmaps.

### LINKED ISSUE #61: Consider not using Show when making GeneralCategory etc.
GeneralCategory.hs is enormous, but this is the nature of having this sort of lookup table in the code. It can be made quite a lot smaller and hopefully ease the work the compiler has to do parsing it by using the numeric values for the chars in it. replacing the generation of `\EOT` with `\4` saves 254,666 characters in the source file, but also has the benefit that a human can look at the table and tell which category fromEnum value a particular position corresponds to. A quick find and replace of some of the more common characters gives this saving in the line length of bitmap#:

```
    Original     -> 3485166
    EOT replaced -> 3230500
    NUL replaced -> 3226838
    NAK replaced -> 3220233
    SOH replaced -> 3215779
    ENQ replaced -> 3211879
    SUB replaced -> 3211716
    ETX replaced -> 3211048
```

I understand this isn't something people are expected to read very often, but for the times when someone does need to, it would be helpful. I'd even consider adding a generated comment which shows the mapping between the numeric and `GeneralCategory` values.

### PR DESCRIPTION
**Pull request by wismill**
Fix based on @axman6-da proposal in #61.

Fixes #61

### CODE REVIEW

File: exe/Parser/Text.hs [hunk 1]
```diff
@@ -178,13 +184,19 @@ __Note:__ 'Enum' instance must respect the following:
...
 * @fromEnum maxBound <= 0xff@
 -}
-enumMapToAddrLiteral :: forall a. (Bounded a, Enum a, Show a) => [a] -> String
-enumMapToAddrLiteral = fmap go
+enumMapToAddrLiteral ::
+  forall a. (Bounded a, Enum a, Show a) =>
+  -- | Values to encode
+  [a] ->
+  -- | String to append
+  String ->
```
Comment (harendra-kumar) [inline exe/Parser/Text.hs:192]: We use the `-> String` style.

**Comment by harendra-kumar** [comment]
I guess this affects only compile time processing and not the generate code. I can see that the file sizes are reduced:

```
(wip/fix_61)$ du -sb *
58788   CombiningClass.hs
54965   Compositions.hs
49987   Decomposable.hs
50960   DecomposableK.hs
63565   Decompositions.hs
70493   DecompositionsK.hs
103510  DecompositionsK2.hs
3207915 GeneralCategory.hs
52590   SimpleLowerCaseMapping.hs
53324   SimpleTitleCaseMapping.hs
53188   SimpleUpperCaseMapping.hs

(master)$ du -sb *
89632   CombiningClass.hs
72909   Compositions.hs
98015   Decomposable.hs
97940   DecomposableK.hs
63565   Decompositions.hs
70493   DecompositionsK.hs
103510  DecompositionsK2.hs
3485793 GeneralCategory.hs
52590   SimpleLowerCaseMapping.hs
53324   SimpleTitleCaseMapping.hs
53188   SimpleUpperCaseMapping.hs
```

GeneralCategory.hs is reduced by around 10% but is still quite large though.

**Review by harendra-kumar — APPROVED** [review]

**Comment by wismill** [comment]
> I guess this affects only compile time processing and not the generate code. I can see that the file sizes are reduced:

Exactly. This is a welcome change!

**Review by adithyaov — APPROVED** [review]
This is nice. LGTM!

**Pull request merged**